learning/tour-of-beam/backend/internal/storage/datastore.go

```diff
@@ -0,0 +1,246 @@
+// Licensed to the Apache Software Foundation (ASF) under one or more
+// contributor license agreements.  See the NOTICE file distributed with
+// this work for additional information regarding copyright ownership.
+// The ASF licenses this file to You under the Apache License, Version 2.0
+// (the "License"); you may not use this file except in compliance with
+// the License.  You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package storage
+
+import (
+	"context"
+	"fmt"
+	"log"
+
+	tob "beam.apache.org/learning/tour-of-beam/backend/internal"
+	"cloud.google.com/go/datastore"
+)
+
+type DatastoreDb struct {
```

Comment:
I recommend you to create interface, e.g Storage where you list all methods for learning materials. Probably, in the future you will want to change the kind of storage

Reply:
There's an interface already, storage.Iface
And also there's a mock storage implementation as well